### ISSUE TITLE
How can I import a function from obfuscated file into an unobfuscated file with the same name before obfuscation

### ISSUE THREAD

**Issue by BrunoTerra97** [javascript-obfuscator/javascript-obfuscator#908]
Hello, I want to import an obfuscated function with the same name before obfuscation

Example:
fileToObfuscate.js export function doSomeThing
someFile.js import doSomeThing from "fileToObfuscate"

Expected:
After obfuscate someFile.js export doSomeThing with the same call as before obfuscation

What happens:
After obfuscating someFile.js fileToObfuscate.js cannot import doSomeThing, because its export has been "changed"

Thanks!

**Comment by sanex3339** [comment]
Hi. Use `reservedNames` option (https://github.com/javascript-obfuscator/javascript-obfuscator#reservednames):

```
reservedNames: ['^doSomeThing$']
```

**Comment by BrunoTerra97** [comment]
@sanex3339 Thanks for answering my question so fast!

After I added
`reservedNames: ['^doSomeThing$']`
My code still doesn't recognize the import. So I started to suspect something in my code and discovered that the problem is with typescript.
What I didn't say before is that the file I'm importing from the obfuscated function is a ts file. Therefore, typescript does not recognize the export of the obfuscated file and says that the function exists, but it is not exported from the obfuscated file.
`Module '"./fileToObfuscate"' declares 'doSomeThing' locally, but it is not exported.`

Therefore, the solution is to create a type definition file before obfuscating the fileToObfuscate.js
After that, the code works perfectly!

Thank you very much for your library!
